cli/cmd/install/install.go

```diff
@@ -0,0 +1,496 @@
+	// is created will use this command's flags first, then the HELM_KUBECONTEXT environment variable,
+	// then call out to genericclioptions.ConfigFlag
+	settings := helmCLI.New()
+	os.Setenv("HELM_NAMESPACE", prevHelmNSEnv)
+
+	if c.flagKubeConfig != "" {
+		settings.KubeConfig = c.flagKubeConfig
+	}
+	if c.flagKubeContext != "" {
+		settings.KubeContext = c.flagKubeContext
+	}
+
+	// Setup logger to stream Helm library logs
+	var uiLogger = func(s string, args ...interface{}) {
+		logMsg := fmt.Sprintf(s, args...)
+		c.UI.Output(logMsg, terminal.WithInfoStyle())
+	}
+
+	// Setup action configuration for Helm Go SDK function calls.
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(settings.RESTClientGetter(), c.flagNamespace,
+		os.Getenv("HELM_DRIVER"), uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error())
+		return 1
+	}
+
+	// Set up the kubernetes client to use for non Helm SDK calls to the Kubernetes API
+	// The Helm SDK will use settings.RESTClientGetter for its calls as well, so this will
+	// use a consistent method to target the right cluster for both Helm SDK and non Helm SDK calls.
+	if c.kubernetes == nil {
+		restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+		if err != nil {
+			c.UI.Output("Retrieving Kubernetes auth: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("Initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	c.UI.Output("Pre-Install Checks", terminal.WithHeaderStyle())
+
+	err = c.checkForPreviousInstallations(settings, uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous PVCs lying around.
+	err = c.checkForPreviousPVCs()
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous bootstrap secret lying around.
+	err = c.checkForPreviousSecrets()
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Handle preset, value files, and set values logic.
+	vals, err := c.mergeValuesFlagsWithPrecedence(settings)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	valuesYaml, err := yaml.Marshal(vals)
+
+	// Print out the installation summary.
+	if !c.flagAutoApprove {
+		c.UI.Output("Consul Installation Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", DefaultReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", c.flagNamespace, terminal.WithInfoStyle())
+
+		if err != nil {
+			c.UI.Output("Overrides:"+"\n"+"%+v", err, terminal.WithInfoStyle())
+		} else if len(vals) == 0 {
+			c.UI.Output("Overrides: "+string(valuesYaml), terminal.WithInfoStyle())
+		} else {
+			c.UI.Output("Overrides:"+"\n"+string(valuesYaml), terminal.WithInfoStyle())
+		}
```

Comment:
This could be the error check for the previous error but it is unclear why it is getting checked here IMO. Also, does it not exit if there was an error marshalling the YAML?

Reply:
You're totally right, the error should be checked as per your previous comment, and not here, updated!